.github/workflows/PR-U20.yaml

```diff
@@ -0,0 +1,127 @@
+        git config user.email "opencv.ci"
+        git config user.name "opencv.ci"
+        git pull -v "https://github.com/${{ env.PR_AUTHOR }}/opencv" "${{ env.SOURCE_BRANCH_NAME }}" --allow-unrelated-histories
+    - name: Clone opencv_extra
+      run: git clone --single-branch --branch ${{ env.TARGET_BRANCH_NAME }} --depth 1 https://github.com/opencv/opencv_extra.git /opencv_extra
+    - name: Configure OpenCV
+      run: |
+        cd /opencv-build
+        cmake -G Ninja ${{ env.EXTRA_CMAKE_OPTIONS }} ${{ env.OPENCV_DOCKER_WORKDIR }}
+    - name: Build OpenCV
+      run: |
+        cd /opencv-build
+        ninja
+    - name: Accuracy
+      run: |
+        cd /opencv-build
+        accuracy_tests=($(ls bin/opencv_test*))
+        for test_name in "${accuracy_tests[@]}"; do
+          if [[ $(basename ${test_name}) != 'opencv_test_viz' ]]; then
+            xvfb-run -a ${test_name}
+          fi
+        done
+    - name: Performance
+      run: |
+        cd /opencv-build
+        performance_tests=($(ls bin/opencv_perf*))
+        for test_name in "${performance_tests[@]}"; do
+          if [[ $(basename ${test_name}) != 'opencv_perf_viz' ]]; then
+            xvfb-run -a ${test_name} --perf_impl=plain --perf_min_samples=1 --perf_force_samples=1 --perf_verify_sanity --skip_unstable=1
+          fi
+        done
+    - name: Python3
+      run: |
+        cd ${{ env.OPENCV_DOCKER_WORKDIR }}/modules/python/test
+        python3 ./test.py --repo ../../../
+    - name: Java
+      run: |
+        cd /opencv-build
+        xvfb-run -a python3 ${{ env.OPENCV_DOCKER_WORKDIR }}/modules/ts/misc/run.py . -a -t java
+    - name: Pylint
+      run: |
+        cd /opencv-build
+        cmake --build . --config release --target check_pylint -- -j4
+
+  B-Contrib:
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: quay.io/asenyaev/opencv-ubuntu:20.04
+    steps:
+    - name: PR info
+      run: |
+        echo "PR Author: ${{ env.PR_AUTHOR }}"
+        echo "Source branch name: ${{ env.SOURCE_BRANCH_NAME }}"
+        echo "Target branch name: ${{ env.TARGET_BRANCH_NAME }}"
+    - name: Clean
+      run: find . -mindepth 1 -delete
+    - name: Fetch opencv
+      uses: actions/checkout@v3
+      with:
+        repository: opencv/opencv
+        ref: ${{ env.TARGET_BRANCH_NAME }}
+        fetch-depth: 0
+    - name: Merge opencv with a test branch
+      run: |
+        cd ${{ env.OPENCV_DOCKER_WORKDIR }}
+        git config --global --add safe.directory ${{ env.OPENCV_DOCKER_WORKDIR }}
+        git config user.email "opencv.ci"
+        git config user.name "opencv.ci"
+        git pull -v "https://github.com/${{ env.PR_AUTHOR }}/opencv" "${{ env.SOURCE_BRANCH_NAME }}" --allow-unrelated-histories
+    - name: Clone opencv_contrib
+      run: git clone --single-branch --branch ${{ env.TARGET_BRANCH_NAME }} --depth 1 https://github.com/opencv/opencv_contrib.git /opencv_contrib
```

Comment:
Should it be tested with PR author's source branch of fork of opencv_contrib if existed?

Reply:
No, there are a bit unstable tests which can affect checks in opencv repository. The opencv_contrib repository will have it's own Github Actions with tests.